tutorials/python/python-type-driven-development/tutorial.md

```diff
@@ -0,0 +1,209 @@
+---
+title: Type-Driven Development in Python
+published: 2019-08-01
+tags: Python, typing, types, tdd
+language: python
+author-name: Carlos Villavicencio
+twitter-profile: po5i
+github-profile: po5i
+description: Learn how to improve the software quality and readability in our Python code by using the built-in typing system and complementing with other software development processes like type-driven development and test-driven development.
+---
+
+_Type-Driven Development_ is a simple approach that states to add static types to our code that are aligned with the logic. It brings many benefits such as increasing the performance, accuracy, testability, readability, and extensibility of your code by taking advantage of the language’s type system. There are many books that you can use to learn how to achieve this, for example, [“Type-Driven Development with Idris”][tdd-idris] and extrapolate these practices to other languages.
+
+In a [previous article][types-vs-tests], we discussed the benefits of working with _Type-Driven Development_ and the importance of static type checking in compilation time for strongly typed languages such as Java or Haskell. It doesn’t matter if the paradigm is object-oriented or functional, type-checking adds a layer of safety, reducing the amount of tests we need to write, and reduces our dependency on tests.
+
```

Comment:
change first "," after functional to a hyphen since the sentence switches a bit more than the comment would indicate at that point.